tests/console/repo_package_install.pm

```diff
@@ -0,0 +1,35 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: check SLE repositories, packages and installation status
+# Maintainer: Zaoliang Luo <zluo@suse.com>
+
+use base "consoletest";
+use strict;
+use testapi;
+use utils;
+
+my @packages = (
+    {
+        name   => 'salt',
+        repo   => 'Basesystem_Module',
+        status => 'not installed'
+    });
+
+sub run {
+    foreach (@packages) {
+        #run zypper command to check repositories, packages and installation status
+        assert_script_run("zypper lr | grep '" . $_->{repo} . "'");
```

Comment:
I think we are cheating ourselves here . From structure above it look like "let's check that package with certain state exists in certain repository" but in a fact it is "let's check two things : 1. certain repository is exists 2. certain package exists and have certain state". 
I see two possible solutions to this problem :
1. Grep for repo in zypper info 
2. Split structure into two - one will contain only repositories and another will contain package+state

Reply:
so you are not agree with the ticket? The test is about to check repo, package and it's installation. nothing about cheating. Sorry I don't get your point here.

Reply:
@asmorodskyi good point, somehow missed it. Yes, with current implementation we do not verify if package is available in defined repo.

Reply:
@okurz package part already added  and pushed:

assert_script_run("zypper info '" . $_->{name} . "' | grep '" . $_->{repo} . "'");